test/files/neg/t13070.scala

```diff
@@ -0,0 +1,10 @@
+//> using options -Wunused:patvars -Werror
+class C {
+  def g = {
+    val t = Option((27, 42))
+    for {
+      ns <- t
+      (i, j) = ns
```

Comment:
let's add a variant to the test where i/j are used

Reply:
from the other compiler, I've forgotten what the behavior is here. I seem to remember people complaining that they can't make just one mutable.
```
object `patvar is assignable`:
  var (i, j) = (42, 27) // no warn nonprivate
  j += 1
  println((i, j))

object `privy patvar is assignable`:
  private var (i, j) = (42, 27) // warn
  j += 1
  println((i, j))

object `local patvar is assignable`:
  def f() =
    var (i, j) = (42, 27) // warn
    j += 1
    println((i, j))

object `mutable patvar in for`:
  def f(xs: List[Int]) =
    for x <- xs; y = x + 1 if y > 10 yield
      var z :: Nil = y :: Nil: @unchecked // warn
      z + 10

class `unset var requires -Wunused`:
  private var i = 0 // no warn as we didn't ask for it
  def f = println(i)
```

Reply:
I guess it doesn't warn for patvar def under patvars but under privates/locals. Even though they are patvars.

Hmm...

Reply:
Maybe that was only because the patvar warning was unreliable, so for maximal utility it relied on the other flag.